app/models/navigation_link.rb

```diff
@@ -1,5 +1,5 @@
 class NavigationLink < ApplicationRecord
-  SVG_REGEXP = /<svg .*>/im
+  SVG_REGEXP = /\A<svg .*>[\s]*\z/im
```

Comment:
One comment about this: the `[\s]*` is needed because basically all svgs in `app/assets/images/twemoji/` has an extra empty line.

IMO: I prefer to remove it (the `[\s]`) from the regexp and change all svgs, but I thought it was better to check with y'all first.

Reply:
I'm personally fine with the looser regex as it's one fewer thing to keep track of whenever we add new SVGs to the repo.

This said, I actually wonder if using `validates_with` is the more "proper" approach here and then calling out to `libmagic` to verify that the document is SVG without using regex (regex against XML is a notorious way to ask for pain and suffering)... for example, the `file(1)` CLI, which IIRC uses `libmagic` on Linux, recognizes the file as SVG for us:

```
(woods) _forem1  » main » file -i app/assets/images/twemoji/tag.svg       
app/assets/images/twemoji/tag.svg: image/svg+xml; charset=us-ascii
```

No real need to change that as part of this PR I think, I'm mostly brainstorming out loud because I agree that regex is a smell, I just think I might disagree with the path forward being "so change our SVGs to fit a tighter regex".

Reply:
From the time I've spent researching this change I could not find a place where these icons are actually being displayed in the UI, that is why I took the looser regex approach instead of cleaning up all svgs.

`ReactionCategory` class is reading the SVGs from the repo, not from the DB.
`app/views/layouts/_nav.html.erb` uses this class but don't display the `icon` attribute.

I do not know much about the history of this feature, but it seems to me these SVGs (from the DB table/column) were used in an earlier version and replaced with texts, but left in the DB.

But `Admin::NavigationLinksController` still allows the icon to make it to DB, so this regexp will prevent nnecessary text/code are injected into the SVG make it to the DB.

Reply:
Ah that's because you are looking in the wrong spot - the `NavigationLink`s are the links rendered in the _sidebar_ (via `app/views/layouts/_sidebar_nav_link.html.erb`). The links (and the icons they show up with) are something that admins of a Forem instance can customize.

If we're touching the validation anyway, then IMO it's very much worth going all the way and checking that it's a proper SVG (especially since execution cost isn't really a concern - people don't change these very often)

Reply:
Using an OS tool that is consistent across the possible dev machines OSes (Mac, Linux, Windows) is difficut and won't add much more than what the regexp does.

For instance, using the example above from @klardotsh :
On my mac, `file` and `file -i` have different behavior:
```
hac@MBP forem % file app/assets/images/twemoji/tag.svg
app/assets/images/twemoji/tag.svg: SVG Scalable Vector Graphics image

hac@MBP forem % file -i app/assets/images/twemoji/tag.svg
app/assets/images/twemoji/tag.svg: regular file
```

Also, on my mac, an svg with .txt extension that includes only the characters `<svg` would return as SVG.
```
hac@MBP forem % file a.txt
a.txt: SVG Scalable Vector Graphics image
hac@MBP forem % cat a.txt
<svg
```

The main goal I've tried to address in this PR is to improve the regular expression to prevent unwanted code to be injected in the SVG, like these tests that I've added in this PR.

Regexp in Ruby that don't use `\A` and `\z` to match beginning and end of its content, is vulnerable (ref: https://guides.rubyonrails.org/security.html#regular-expressions)